### PR TITLE
Add shadowing options for the worker shadowing mode

### PR DESCRIPTION
**Pull request by yux0**
Add shadowing options for the worker shadowing mode.

### CODE REVIEW

File: src/main/java/com/uber/cadence/worker/ShadowingExitCondition.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
+/*
+ *  Modifications Copyright (c) 2017-2020 Uber Technologies Inc.
```
Comment (yycptt) [inline src/main/java/com/uber/cadence/worker/ShadowingExitCondition.java:2]: nit: 2021

File: src/main/java/com/uber/cadence/worker/ShadowingOptions.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+            Objects.requireNonNull(workflowStatuses);
+            this.workflowStatuses = Sets.newHashSet(workflowStatuses);
+            if (workflowStatuses.size() == 0) {
+                this.workflowStatuses.add(WorkflowStatus.OPEN);
+            }
+            return this;
+        }
+
+        public Builder setWorkflowSamplingRate(double samplingRate) {
+            if (samplingRate < 0.0 || samplingRate > 1.0) {
```
Comment (yycptt) [inline src/main/java/com/uber/cadence/worker/ShadowingOptions.java:94]: nit: shall we also treat 0 as invalid?

File: src/main/java/com/uber/cadence/worker/TimeFilter.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        return DEFAULT_INSTANCE;
+    }
+
+    private static final TimeFilter DEFAULT_INSTANCE;
+
+    static {
+        DEFAULT_INSTANCE = TimeFilter.newBuilder().build();
+    }
+
+    public static final class Builder {
```
Comment (yycptt) [inline src/main/java/com/uber/cadence/worker/TimeFilter.java:41]: nit: I am not sure if having a Builder for every option field is an overkill or not. But I am not familiar with java client, so feel free to ignore this comment.

**Review by yycptt — APPROVED** [review]
LGTM. 

We may need to think about how to reuse part of the code when defining the options for shadowing worker. I guess it will be very similar to this option for local shadowing?

File: src/main/java/com/uber/cadence/worker/ShadowingOptions.java (hunk 2 above)
Comment (yux0) [reply src/main/java/com/uber/cadence/worker/ShadowingOptions.java]: Yes. Updated

File: src/main/java/com/uber/cadence/worker/TimeFilter.java (hunk 3 above)
Comment (yux0) [reply src/main/java/com/uber/cadence/worker/TimeFilter.java]: Had the same feeling. But I incline to keep the consistency here.

**Pull request merged**